apps/prairielearn/assets/scripts/lib/countdown.ts

```diff
@@ -88,15 +96,16 @@ export function setupCountdown(options: {
     }
   }
 
-  function updateServerRemainingMS() {
+  function updateServerRemainingMS(lastRemainingMS: number) {
```

Comment:
This is called by `setInterval`, which won't invoke it with a value for `lastRemainingMS`. The types here and downstream don't reflect that.

Can you way more about why `lastRemainingMS` is needed, and ideally document that inline?

Reply:
The parameter is passed onto `onServerUpdateFail` in case of an error. It is currently used in the exam time limit countdown to distinguish the case of an update failing before or after `lastRemainingMS` reaches 0. On the workspace page, instead of reloading the page a warning message is displayed to students for failures with `lastRemainingMS > 0`, and the error is ignored otherwise (since it'll be triggered intentionally by the assignment being closed).

Reply:
I've changed this slightly to directly calculate `remainingMS` in `updateServerRemainingMS`, so that the signature of that function doesn't need to be changed.